src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1752,18 +1768,18 @@ static void glb_update_locked(grpc_exec_ctx* exec_ctx, grpc_lb_policy* policy,
           "glb_update_missing");
     } else {
       // otherwise, keep using the current LB channel (ignore this update).
-      gpr_log(GPR_ERROR,
-              "No valid LB addresses channel arg for grpclb %p update, "
-              "ignoring.",
-              (void*)glb_policy);
+      gpr_log(
+          GPR_ERROR,
+          "[grpclb %p] No valid LB addresses channel arg in update, ignoring.",
+          glb_policy);
     }
     return;
   }
   const grpc_lb_addresses* addresses =
       (const grpc_lb_addresses*)arg->value.pointer.p;
   // If a non-empty serverlist hasn't been received from the balancer,
   // propagate the update to fallback_backend_addresses.
-  if (glb_policy->serverlist == NULL) {
+  if (glb_policy->started_picking && glb_policy->serverlist == NULL) {
```

Comment:
I don't think this is correct.

Two things happen inside of `fallback_update_locked()`:
1. We update `glb_policy->fallback_backend_addresses` with the new list of backends provided by the resolver.
2. If we are already in fallback mode, we pass that updated list of addresses to the RR policy by calling `rr_handover_locked()`.

Note that even if we have not yet started picking, we still need to do (1), so that when we do start picking later, we will use the right set of backend addresses as the fallback.  However, we do not want (2) to happen in this case.  So I think the bug here is that `fallback_update_locked()` is using the wrong condition to decide when to call `rr_handover_locked()`.  This fix should be made there instead of here.

Note that in #12587, I suggested that the condition in `fallback_update_locked()` should actually be `glb_policy->rr_policy != NULL`.  However, @AspirinSJL responded correctly that that causes problems when the RR policy shuts down due to all of the backends going away.  That problem will go away with #12829, so if that PR gets merged before this one, we can just check that.

Reply:
I see. I'm dropping this PR. I've separated the other two concerns in https://github.com/grpc/grpc/pull/13332 (logging) and https://github.com/grpc/grpc/pull/13331 (retry timer).

Reply:
I've filed https://github.com/grpc/grpc/issues/13333 to track this.

Reply:
Perhaps I was unclear.  I don't think we need a separate issue for this, nor do I think we need to wait for #12829 to go in first.  I think we can just change `fallback_update_locked()` from this:

```
   if (glb_policy->lb_fallback_timeout_ms > 0 &&
       !glb_policy->fallback_timer_active) {
```

to this:

```
   if (glb_policy->started_picking &&
       glb_policy->lb_fallback_timeout_ms > 0 &&
       !glb_policy->fallback_timer_active) {
```

I think we can go ahead and do that immediately, without waiting for #12829.

My comment was simply that once #12829 goes in, the condition in `fallback_update_locked()` can change from what's shown above to simply `glb_policy->rr_policy != NULL`.

Reply:
I have appended a commit in #12829 to change the condition to "glb_policy->rr_policy != NULL".